modules/ztp-creating-siteconfig-custom-resources.adoc

```diff
@@ -0,0 +1,282 @@
+
+
+.Procedure
+
+. Create a ClusterImageSet for each specific cluster version that needs to be deployed. A ClusterImageSet has the following format:
++
+[source,yaml]
+----
+apiVersion: hive.openshift.io/v1
+kind: ClusterImageSet
+metadata:
+  name: openshift-4.8.0-rc.0 <1>
+spec:
+   releaseImage: quay.io/openshift-release-dev/ocp-release:4.8.0-x86_64 <2>
+----
+<1> `name` is the descriptive version that you want to deploy.
+<2> `releaseImage` needs to point to the specific release image to deploy.
+
+
+. Create the namespace for the single node managed cluster:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+     name: <cluster-name> <1>
+     labels:
+        name: <cluster-name> <1>
+----
+<1>  `cluster-name` is the name of the managed single node cluster to provision.
+
+. Create the BMC secret custom resource:
++
+[source,yaml]
+----
+apiVersion: v1
+data:
+  password: <bmc-password> <1>
+  username: <bmc-username> <2>
+kind: Secret
+metadata:
+  name: <cluster-name>-bmc-secret
+  namespace: <cluster-name>
+type: Opaque
+----
+<1> `bmc-password` is the password to the target bare metal machine. Must be base-64 encoded.
+<2> `bmc-username` is the username to the target bare metal machine. Must be base-64 encoded.
+
+. Create the Image Pull Secret custom resource:
++
+[source,yaml]
+----
+apiVersion: v1
+data:
+  .dockerconfigjson: <pull-secret> <1>
+kind: Secret
+metadata:
+  name: assisted-deployment-pull-secret
+  namespace: <cluster-name>
+type: kubernetes.io/dockerconfigjson
+----
+<1> `pull-secret` is the {product-title} pull secret. Must be base-64 encoded.
+
+. Create the AgentClusterInstall custom resource:
++
+[source,yaml]
+----
+apiVersion: extensions.hive.openshift.io/v1beta1
+kind: AgentClusterInstall
+metadata:
+  # Only include the annotation if using OVN, otherwise omit the annotation
+  annotations:
+    agent-install.openshift.io/install-config-overrides: '{"networking":{"networkType":"OVNKubernetes"}}'
+  name: <cluster-name>
+  namespace: <cluster-name>
+spec:
+  clusterDeploymentRef:
+    name: <cluster-name>
+  imageSetRef:
+    name: <cluster-image-set> <1>
+  networking:
+    clusterNetwork:
+    - cidr: 10.128.0.0/14
+      hostPrefix: 23
+    machineNetwork:
+    - cidr: <machine-network-cidr> <2>
+    serviceNetwork:
+    - 172.30.0.0/16
+  provisionRequirements:
+    controlPlaneAgents: 1
+    workerAgents: 0
+  sshPublicKey: <public-key> <3>
+----
++
+<1> `cluster-image-set` is the name of the ClusterImageSet custom resource.
+<2> `machine-network-cidr` is the target bare metal machine’s CIDR.
+<3> `public-key` entered as plain text can be used to SSH into the target bare metal machine after the host is installed.
++
+[NOTE]
+====
+If you want to configure a static IP for the managed cluster at this point, see the procedure in this document for configuring static IP addresses for managed clusters.
```

Comment:
Do you ever want to create it at a different point? If this is the only time that the user will want to create that resource, consider changing that module to optional step(s) here.

Reply:
@StephenJamesSmith, this one, too.

Reply:
Please make sure that you do clarify this after GA - it's not the best to break up the workflow like this.